inline-c-cpp/cxx-src/HaskellException.cxx

```diff
@@ -46,25 +46,41 @@ const char* HaskellException::what() const noexcept {
 // <https://stackoverflow.com/questions/561997/determining-exception-type-after-the-exception-is-caught/47164539#47164539>
 // regarding how to show the type of an exception.
 
+/* mallocs a string representing the exception type name or error condition.
+
+   Ideally, this returns a demangled string, but it may degrade to
+    - a mangled string if demangling fails,
+    - "<unknown exception>" if exception type info is not available,
+    - "<no exception>" if no current exception is found.
+ */
 #if defined(__GNUC__) || defined(__clang__)
 const char* currentExceptionTypeName()
 {
+  std::type_info *type_info = abi::__cxa_current_exception_type();
+  if (!type_info)
+    return strdup("<no exception>");
```

Comment:
This is unrelated to the PR (it was like this to begin with), but could you update the comment in Exception.hs referencing `free`, while it's actually using `unsafePackMallocCString`, and also note here that these allocations are freed there?

Reply:
They do, thank you